doc/api/url.md

```diff
@@ -2,139 +2,247 @@
+The `host` property is the full lower-cased host portion of the URL, including
+the `port` if specified.
 
-* `port`: The port number portion of the host.
+For example: `'host.com:8080'`
 
-    Example: `'8080'`
+### urlObject.auth
 
-* `pathname`: The path section of the URL, that comes after the host and
-  before the query, including the initial slash if present. No decoding is
-  performed.
+The `auth` property is the username and password portion of the URL, also
+referred to as "userinfo". This string subset follows the `protocol` and
+double slashes (if present) and preceeds the `host` component, delimited by an
+ASCII "at sign" (`@`). The format of the string is `{username}[:{password}]`,
+with the `[:{password}]` portion being optional.
 
-    Example: `'/p/a/t/h'`
+For example: `'user:pass'`
 
-* `search`: The 'query string' portion of the URL, including the leading
-  question mark.
+### urlObject.hostname
 
-    Example: `'?query=string'`
+The `hostname` property is the lower-cased host name portion of the `host`
+component *without* the `port` included.
 
-* `path`: Concatenation of `pathname` and `search`. No decoding is performed.
+For example: `'host.com'`
 
-    Example: `'/p/a/t/h?query=string'`
+### urlObject.port
 
-* `query`: Either the 'params' portion of the query string, or a
-  querystring-parsed object.
+The `port` property is the numeric port portion of the `host` component.
 
-    Example: `'query=string'` or `{'query':'string'}`
+For example: `'8080'`
 
-* `hash`: The 'fragment' portion of the URL including the pound-sign.
+### urlObject.pathname
 
-    Example: `'#hash'`
+The `pathname` property consists of the entire path section of the URL. This
+is everything following the `host` (including the `port`) and before the start
+of the `query` or `hash` components, delimited by either the ASCII question
+mark (`?`) or hash (`#`) characters.
 
-### Escaped Characters
+For example `'/p/a/t/h'`
 
-Spaces (`' '`) and the following characters will be automatically escaped in the
-properties of URL objects:
+No decoding of the path string is performed.
 
-```
-< > " ` \r \n \t { } | \ ^ '
-```
+### urlObject.search
+
+The `search` property consists of the entire "query string" portion of the
+URL, including the leading ASCII question mark (`?`) character.
+
+For example: `'?query=string'`
+
+No decoding of the query string is performed.
+
+### urlObject.path
+
+The `path` property is a concatenation of the `pathname` and `search`
+components.
+
+For example: `'/p/a/t/h?query=string'`
+
+No decoding of the `path` is performed.
+
+### urlObject.query
+
+The `query` property is either the "params" portion of the query string (
+everything *except* the leading ASCII question mark (`?`), or an object
+returned by the [`querystring`][] module's `parse()` function:
 
----
+For example: `'query=string'` or `{'query': 'string'}`
 
-The following methods are provided by the URL module:
+No decoding of the query string is performed.
```

Comment:
This needs to clarified to say that this is only the case when the query string isn't parsed. When it is parsed, keys and values are decoded.